src/Type.h

```diff
@@ -623,6 +623,7 @@ class TypeDecl final {
     TypePtr type;
     detail::AttributesPtr attrs;
     const char* id = nullptr;
+    detail::Location loc = detail::GetCurrentLocation();
```

Comment:
Could you instead extend the non-default `TypeDecl` constructor a parameter and default it to to the current location? Mostly thinking that it'd be nice to have some way to controlthe location of a new `TypeDecl` even if you don't have a direct use.

Also, making the member private and adding a `GetLocationInfo()` returning a pointer to it would make it consistent with `class Obj`, which would look more streamlined when locations from either type are used / compared.

Reply:
Making it private and adding `GetLocationInfo()` sounds good. Adding a constructor parameter seems a bit crowded to me given it already has a default parameter, but how about instead adding `SetLocationInfo()` to further reflect the `Obj` interface?

Reply:
> Adding a constructor parameter seems a bit crowded to me given it already has a default parameter, but how about instead adding SetLocationInfo() to further reflect the Obj interface?

Fair. Don't worry about `SetLocationInfo()` then and use the default initializer as you have :+1: If needed, can add it later, too.